src/Ghcid.hs

```diff
@@ -433,11 +436,29 @@ runGhcid session waiter termSize termOutput opts@Options{..} = do
 
 
 -- | Given an available height, and a set of messages to display, show them as best you can.
-prettyOutput :: String -> Int -> [Load] -> [EvalResult] -> [String]
-prettyOutput currTime loadedCount [] evals =
+-- The boolean determines whether file paths in the warning/error messages
+-- will be replaced by absolute paths provided in the loadFile field.
+-- False ~ "keep the exact output from ghci" and
+-- True  ~ "paths will be replaced by absolute paths to help downstream tooling"
+prettyOutput :: Bool -> String -> Int -> [Load] -> [EvalResult] -> [String]
+prettyOutput _replacePaths currTime loadedCount [] evals =
     (allGoodMessage ++ " (" ++ show loadedCount ++ " module" ++ ['s' | loadedCount /= 1] ++ ", at " ++ currTime ++ ")")
         : concatMap printEval evals
-prettyOutput _ _ xs evals = concatMap loadMessage xs ++ concatMap printEval evals
+prettyOutput replacePaths _ _ xs evals =
+  messageLines ++ concatMap printEval evals
+ where
+  messageLines =
+    [ case l of
+        _ | not replacePaths -> l
+        ""                   -> l
+        c : _ | isSpace c    -> l
+        _                    -> if "hs:" `isInfixOf` l
```

Comment:
So this looks for `Foo.hs:` where `hs:` is the suffix of that string, and replaces it with `loadFile`? What if you don't have loadFile as the absolute version of this file, but it's a different file? That seems a dangerous operation to do.